clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainExtensionHeader.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.offheap;
+
+import org.terracotta.offheapstore.storage.portability.WriteContext;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Support extending header and allowing to read the extended area
+ */
+public interface OffHeapChainExtensionHeader {
+  ByteBuffer getExtensionHeader(long xtnAddress);
+  WriteContext getExtensionWriteContext(long xtnAddress);
+  long chainAddressToExtensionAddress(long chainAddress);
```

Comment:
This (and the reverse) shouldn't need to be exposed outside of the OffHeapChainStorageEngine right?